Tools/clinic/clinic.py

```diff
@@ -2589,6 +2623,7 @@ class Parameter:
     annotation: object = inspect.Parameter.empty
     docstring: str = ''
     group: int = 0
+    deprecated_positional: str = ""
```

Comment:
Conceptually, we're in a "different kind of state" if positional usage of the parameter is deprecated. So, while `| None` unions are often annoying, here I think we can use them "for good" -- to signify that there's no deprecation:

```suggestion
    # (`None` signifies that there is no deprecation)
    deprecated_positional: str | None = None
```

Reply:
That's a good idea indeed. I like that.

Reply:
Applied in 9eab8f71c36ce4417a87d98d979661dfa8c98ed0